### PR TITLE
Rename Swap Params / Fix broken links

### PR DESCRIPTION
**Pull request by EndymionJkb**
There are a lot more broken links. Somebody should go through all these files...

I'm trying to keep it current with monorepo changes as far as naming and code snippets, but the code changes have also shifted any specific line references, and there are lots of other broken links. For instance, "html" files all got renamed to "md".

### CODE REVIEW

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 1]
```diff
@@ -19,9 +19,9 @@ At a high level, creating a custom AMM on Balancer protocol involves the impleme
 To expedite the development process, Balancer provides two contracts to inherit from:
 
 - [IBasePool.sol](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/interfaces/contracts/vault/IBasePool.sol) - This interface defines the required functions that every Balancer pool must implement
-- [BalancerPoolToken.sol](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/vault/contracts/BalancerPoolToken.sol) - This contract implements the [ERC20MultiToken](/concepts/vault/erc20-multi-token.html) standard that enables your pool contract to be ERC20 compliant while delegating BPT accounting to the vault. For more information, refer to [BalancerPoolToken](/concepts/core-concepts/balancer-pool-tokens.html).
+- [BalancerPoolToken.sol](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/vault/contracts/BalancerPoolToken.sol) - This contract implements the [ERC20MultiToken](/docs/concepts/vault/erc20-multi-token.md) standard that enables your pool contract to be ERC20 compliant while delegating BPT accounting to the vault. For more information, refer to [BalancerPoolToken](/docs/concepts/core-concepts/balancer-pool-tokens.md).
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
- [BalancerPoolToken.sol](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/vault/contracts/BalancerPoolToken.sol) - This contract implements the [ERC20MultiToken](/docs/concepts/vault/erc20-multi-token.html) standard that enables your pool contract to be ERC20 compliant while delegating BPT accounting to the vault. For more information, refer to [BalancerPoolToken](/docs/concepts/core-concepts/balancer-pool-tokens.html).
```

Comment (mkflow27) [reply docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]: referencing the markdown files does not redirect me to a page on the browser.

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 2]
```diff
@@ -207,15 +207,15 @@ contract ConstantSumPool is IBasePool, BalancerPoolToken {
 ::: info What does Scaled18 mean?
 Internally, Balancer protocol scales all tokens to 18 decimals to minimize the potential for errors that can occur when
 comparing tokens with different decimals numbers (ie: WETH/USDC). `Scaled18` is a suffix used to signify values has already been scaled.
-**By default, ALL values provided to the pool will always be `Scaled18`.** Refer to [Decimal scaling](/concepts/vault/token-scaling.html#pool-registration) for more information.
+**By default, ALL values provided to the pool will always be `Scaled18`.** Refer to [Decimal scaling](/docs/concepts/vault/token-scaling.md#pool-registration) for more information.
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
**By default, ALL values provided to the pool will always be `Scaled18`.** Refer to [Decimal scaling](/docs/concepts/vault/token-scaling.html#pool-registration) for more information.
```

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 3]
```diff
@@ -207,15 +207,15 @@ contract ConstantSumPool is IBasePool, BalancerPoolToken {
...
 :::
 
 ::: info What does Live refer to in balancesLiveScaled18?
-They keyword `Live` denote balances that have been scaled by their respective `IRateProvider` and have any pending yield fee removed. Refer to [Live Balances](/concepts/vault/token-scaling.html#live-balances) for more information.
+They keyword `Live` denote balances that have been scaled by their respective `IRateProvider` and have any pending yield fee removed. Refer to [Live Balances](/docs/concepts/vault/token-scaling.md#live-balances) for more information.
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
They keyword `Live` denote balances that have been scaled by their respective `IRateProvider` and have any pending yield fee removed. Refer to [Live Balances](/docs/concepts/vault/token-scaling.html#live-balances) for more information.
```

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 4]
```diff
@@ -207,15 +207,15 @@ contract ConstantSumPool is IBasePool, BalancerPoolToken {
...
 :::
 
 ::: info How are add and remove liquidity operations implemented?
-Balancer protocol leverages a novel approximation, termed the [Liquidity invariant approximation](/concepts/vault/liquidity-invariant-approximation.html), to provide a generalized solution for liquidity operations.
+Balancer protocol leverages a novel approximation, termed the [Liquidity invariant approximation](/docs/concepts/vault/liquidity-invariant-approximation.md), to provide a generalized solution for liquidity operations.
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
Balancer protocol leverages a novel approximation, termed the [Liquidity invariant approximation](/docs/concepts/vault/liquidity-invariant-approximation.html), to provide a generalized solution for liquidity operations.
```

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 5]
```diff
@@ -258,11 +258,11 @@ Our two-token `ConstantSumPool` uses the constant sum invariant, or `X + Y = K`.
...
-- bptAmountOut for an unbalanced add liquidity operation should equal bptAmountOut for a proportional add liquidity in the case that `exactAmountsIn` for the unbalanced add are equal to the `amountsIn` for the same bptAmountOut for both addLiquidity scenarios. `AddLiquidityProportional` does not call into the custom pool it instead calculates BptAmountOut within the [BasePoolMath.sol](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/vault/contracts/BasePoolMath.sol#L50-L71) whereas `addLiquidityUnbalanced` calls the custom pool's `computeInvariant`. 
+- bptAmountOut for an unbalanced add liquidity operation should equal bptAmountOut for a proportional add liquidity in the case that `exactAmountsIn` for the unbalanced add are equal to the `amountsIn` for the same bptAmountOut for both addLiquidity scenarios. `AddLiquidityProportional` does not call into the custom pool it instead calculates BptAmountOut using [BasePoolMath.sol](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/solidity-utils/contracts/math/BasePoolMath.sol#L50-L54) whereas `addLiquidityUnbalanced` calls the custom pool's `computeInvariant`.
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
- bptAmountOut for an unbalanced add liquidity operation should equal bptAmountOut for a proportional add liquidity in the case that `exactAmountsIn` for the unbalanced add are equal to the `amountsIn` for the same bptAmountOut for both addLiquidity scenarios. `AddLiquidityProportional` does not call into the custom pool it instead calculates BptAmountOut using [BasePoolMath.sol](https://github.com/balancer/balancer-v3-monorepo/blob/ba8035727441b04bea463b3e9cc614d91395bcf8/pkg/vault/contracts/BasePoolMath.sol#L50-L54) whereas `addLiquidityUnbalanced` calls the custom pool's `computeInvariant`. 
```

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 6]
```diff
@@ -321,7 +321,7 @@ Balancer protocol supports two types of swaps:
 - `EXACT_IN` - The user defines the exact amount of `tokenIn` they want to spend.
 - `EXACT_OUT` - The user defines the exact amount of `tokenOut` they want to receive.
 
-The `minAmountOut` or `maxAmountIn` are enforced by the [vault](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/vault/contracts/Vault.sol#L405) .
+The `minAmountOut` or `maxAmountIn` are enforced by the [vault](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/vault/contracts/Vault.sol#L391-L393) .
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]: @EndymionJkb You know the vault in your sleep. I would need to read through it to find that check again. Which lines are you refering to here?

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 7]
```diff
@@ -378,16 +377,16 @@ No, swap fees are managed entirely by the Balancer vault. For an `EXACT_OUT` swa
 Balancer supports two types of swap fees:
 
 - **Static swap fee**: Defined on `vault.registerPool()` and managed via calls to `vault.setStaticSwapFeePercentage()`. For more information, see [Swap fee](/concepts/vault/swap-fee.html).
-- **Dynamic swap fee**: are managed by a **Hooks** contract. If a swap with a pool uses the dynamic swap fee is determined on pool registration. A Hook flags that it supports dynamic fees on `vault.registerPool()`. For more information, see [Dynamic swap fees](/concepts/vault/swap-fee.html#dynamic-swap-fee).
+- **Dynamic swap fee**: are managed by a **Hooks** contract. If a swap with a pool uses the dynamic swap fee is determined on pool registration. A Hook flags that it supports dynamic fees on `vault.registerPool()`. For more information, see [Dynamic swap fees](/docs/concepts/vault/swap-fee.md#dynamic-swap-fee).
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
- **Dynamic swap fee**: are managed by a **Hooks** contract. If a swap with a pool uses the dynamic swap fee is determined on pool registration. A Hook flags that it supports dynamic fees on `vault.registerPool()`. For more information, see [Dynamic swap fees](/docs/concepts/vault/swap-fee.html#dynamic-swap-fee).
```

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 8]
```diff
@@ -378,16 +377,16 @@ No, swap fees are managed entirely by the Balancer vault. For an `EXACT_OUT` swa
...
 
 ## Hooks
-Hooks as standalone contracts are not part of a custom pool's implementation. However they can be combined with custom pools. For a detailed understanding, see [Hooks](/concepts/core-concepts/hooks.html).
+Hooks as standalone contracts are not part of a custom pool's implementation. However they can be combined with custom pools. For a detailed understanding, see [Hooks](/docs/concepts/core-concepts/hooks.md).
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
Hooks as standalone contracts are not part of a custom pool's implementation. However they can be combined with custom pools. For a detailed understanding, see [Hooks](/docs/concepts/core-concepts/hooks.html).
```

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 9]
```diff
@@ -378,16 +377,16 @@ No, swap fees are managed entirely by the Balancer vault. For an `EXACT_OUT` swa
...
 
 ### Vault reentrancy
 Hooks allow a pool to reenter the vault within the context of a pool operation. While `onSwap`, `computeInvariant` and `computeBalance` must be executed within a reentrancy guard, the vault is architected such that hooks operate outside of this requirement.
 
 ## Add / Remove liquidity 
-The implementation of `computeInvariant` and `computeBalance` allows a pool to support ALL [Add/Remove liquidity types](/concepts/vault/add-remove-liquidity-types.html).
+The implementation of `computeInvariant` and `computeBalance` allows a pool to support ALL [Add/Remove liquidity types](/docs/concepts/vault/add-remove-liquidity-types.md).
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
The implementation of `computeInvariant` and `computeBalance` allows a pool to support ALL [Add/Remove liquidity types](/docs/concepts/vault/add-remove-liquidity-types.html).
```

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 10]
```diff
@@ -430,17 +429,17 @@ struct LiquidityManagement {
     bool enableDonation;
 }
 ```
-These settings get passed into the [pool registration](/developer-reference/contracts/vault-api.html#registerpool) flow.
+These settings get passed into the [pool registration](/docs/developer-reference/contracts/vault-api.md#registerpool) flow.
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
These settings get passed into the [pool registration](/docs/developer-reference/contracts/vault-api.html#registerpool) flow.
```

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md [hunk 11]
```diff
@@ -430,17 +429,17 @@ struct LiquidityManagement {
...
 
 
 ## Deploying your pool
 
-See the guide to [Deploy a Custom AMM Using a Factory](/build-a-custom-amm/build-an-amm/deploy-custom-amm-using-factory.html).
+See the guide to [Deploy a Custom AMM Using a Factory](/docs/build-a-custom-amm/build-an-amm/deploy-custom-amm-using-factory.md).
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]:
```suggestion
See the guide to [Deploy a Custom AMM Using a Factory](/docs/build-a-custom-amm/build-an-amm/deploy-custom-amm-using-factory.html).
```

File: docs/build-a-custom-amm/build-an-amm/deploy-custom-amm-using-factory.md [hunk 12]
```diff
@@ -5,7 +5,7 @@ title: Deploy a Custom AMM Using a Factory
 
 # Deploy a Custom AMM Using a Factory
 
-_This section is for developers looking to deploy a custom pool contract that has already been written. If you are looking to design a custom AMM with a novel invariant, start [here](/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.html)._
+_This section is for developers looking to deploy a custom pool contract that has already been written. If you are looking to design a custom AMM with a novel invariant, start [here](/docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md)._
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/deploy-custom-amm-using-factory.md]:
```suggestion
_This section is for developers looking to deploy a custom pool contract that has already been written. If you are looking to design a custom AMM with a novel invariant, start [here](/docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.html)._
```

File: docs/build-a-custom-amm/build-an-amm/deploy-custom-amm-using-factory.md [hunk 13]
```diff
@@ -26,7 +26,7 @@ To see example foundry scripts for deploying a custom pool using a factory, chec
 
 A factory contract should inherit the [BasePoolFactory.sol](https://github.com/balancer/balancer-v3-monorepo/blob/main/pkg/pool-utils/contracts/BasePoolFactory.sol) abstract contract, which sets the table for deploying pools with `CREATE3` and streamlines the registration process.
 
-Below, we present an example custom pool factory that uses the `ConstantSumPool` contract from [Build your custom AMM](/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.html#build-your-custom-amm)
+Below, we present an example custom pool factory that uses the `ConstantSumPool` contract from [Build your custom AMM](/docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md#build-your-custom-amm)
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/deploy-custom-amm-using-factory.md]:
```suggestion
Below, we present an example custom pool factory that uses the `ConstantSumPool` contract from [Build your custom AMM](/docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.html#build-your-custom-amm)
```

File: docs/build-a-custom-amm/build-an-amm/extend-existing-pool-type-using-hooks.md [hunk 14]
```diff
@@ -5,12 +5,12 @@ title: Extend an Existing Pool Type Using Hooks
 
 # Extend an Existing Pool Type Using Hooks
 
-_This section is for developers looking to extend an existing pool type with custom hooks. If you are looking to create a custom AMM with a novel invariant, start [here](/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.html)._
+_This section is for developers looking to extend an existing pool type with custom hooks. If you are looking to create a custom AMM with a novel invariant, start [here](/docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md)._
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/extend-existing-pool-type-using-hooks.md]:
```suggestion
_This section is for developers looking to extend an existing pool type with custom hooks. If you are looking to create a custom AMM with a novel invariant, start [here](/docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.html)._
```

File: docs/build-a-custom-amm/build-an-amm/extend-existing-pool-type-using-hooks.md [hunk 15]
```diff
@@ -5,12 +5,12 @@ title: Extend an Existing Pool Type Using Hooks
...
 
 Hooks introduce a new framework for extending the functionality of existing pool types at key points throughout their lifecycle. By enabling actions during pool operations and facilitating dynamic swap fee computation, hooks offer unprecedented control over pool behavior. This innovative concept empowers developers to craft tailored pool behaviors, catering to specific use cases and enhancing operations with greater flexibility and control.
 
 ::: info
-Before you start with this walkthrough, consider reading through the [technical section on hooks](/concepts/core-concepts/hooks.html#hook-contracts) and take a look at the [Hooks API](/developer-reference/contracts/hooks-api.html).
+Before you start with this walkthrough, consider reading through the [technical section on hooks](/docs/concepts/core-concepts/hooks.md#hook-contracts) and take a look at the [Hooks API](/docs/developer-reference/contracts/hooks-api.md).
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/extend-existing-pool-type-using-hooks.md]:
```suggestion
Before you start with this walkthrough, consider reading through the [technical section on hooks](/docs/concepts/core-concepts/hooks.md#hook-contracts) and take a look at the [Hooks API](/docs/developer-reference/contracts/hooks-api.html).
```

File: docs/build-a-custom-amm/build-an-amm/extend-existing-pool-type-using-hooks.md [hunk 16]
```diff
@@ -154,5 +154,5 @@ function onComputeDynamicSwapFeePercentage(
 
 Now we can implement the logic in the `onComputeDynamicSwapFeePercentage` function, which the Vault calls to retrieve the swap fee value. In our example, any veBal holder enjoys a 50% swap fee discount, instead of the default static swap fee. However, there are some nuances to consider in this implementation.
 
-To obtain the user's veBAL balance, we need the sender's address, which we can retrieve by calling `getSender()` on the router. This relies on the router returning the correct address, so it's crucial to ensure the router is "trusted" (any contract can act as a [Router](/concepts/router/overview.html#routers)). In our example we passed a trusted `_router` address, which is saved during the hook deployment.
+To obtain the user's veBAL balance, we need the sender's address, which we can retrieve by calling `getSender()` on the router. This relies on the router returning the correct address, so it's crucial to ensure the router is "trusted" (any contract can act as a [Router](/docs/concepts/router/overview.md#routers)). In our example we passed a trusted `_router` address, which is saved during the hook deployment.
```
Comment (mkflow27) [inline docs/build-a-custom-amm/build-an-amm/extend-existing-pool-type-using-hooks.md]:
```suggestion
To obtain the user's veBAL balance, we need the sender's address, which we can retrieve by calling `getSender()` on the router. This relies on the router returning the correct address, so it's crucial to ensure the router is "trusted" (any contract can act as a [Router](/docs/concepts/router/overview.html#routers)). In our example we passed a trusted `_router` address, which is saved during the hook deployment.
```

File: docs/concepts/core-concepts/architecture.md [hunk 17]
```diff
@@ -21,9 +21,9 @@ The following diagram illustrates the usage of each component during a transacti
 
 ![Router Vault interaction](/images/architecture-simplified.png)
 
-1. The [Router](/concepts/router/technical.html) acts as the primary interface for accessing the Balancer protocol, offering a user-friendly way to interact with the Balancer Vault.
+1. The [Router](/docs/concepts/router/technical.md) acts as the primary interface for accessing the Balancer protocol, offering a user-friendly way to interact with the Balancer Vault.
```
Comment (mkflow27) [inline docs/concepts/core-concepts/architecture.md]:
```suggestion
1. The [Router](/docs/concepts/router/technical.md) acts as the primary interface for accessing the Balancer protocol, offering a user-friendly way to interact with the Balancer Vault.
```
```suggestion
1. The [Router](/docs/concepts/router/technical.html) acts as the primary interface for accessing the Balancer protocol, offering a user-friendly way to interact with the Balancer Vault.
```

File: docs/concepts/core-concepts/hooks.md [hunk 18]
```diff
@@ -126,7 +126,7 @@ A detailed view of what an `after` hook for a given liquidity operation can chan
 
 
 ## Hook examples
-If you want to get started with developing your own hooks contract, check out the [developing a hooks contract](/build-a-custom-amm/build-an-amm/extend-existing-pool-type-using-hooks.html) page. Various hook examples are shown there. Additionally the monorepo displays more ideas on how to approach hook development.
+If you want to get started with developing your own hooks contract, check out the [developing a hooks contract](/docs/build-a-custom-amm/build-an-amm/extend-existing-pool-type-using-hooks.md) page. Various hook examples are shown there. Additionally the monorepo displays more ideas on how to approach hook development.
```
Comment (mkflow27) [inline docs/concepts/core-concepts/hooks.md]:
```suggestion
If you want to get started with developing your own hooks contract, check out the [developing a hooks contract](/docs/build-a-custom-amm/build-an-amm/extend-existing-pool-type-using-hooks.html) page. Various hook examples are shown there. Additionally the monorepo displays more ideas on how to approach hook development.
```

**Review by mkflow27 — CHANGES_REQUESTED** [review]
Thanks @EndymionJkb I made suggestions to change the references to the md files to be html again, otherwise the link does not lead to a page in the browser version of the docs.

File: docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md (hunk 6 above)
Comment (EndymionJkb) [reply docs/build-a-custom-amm/build-an-amm/create-custom-amm-with-novel-invariant.md]: Missed that link. I'll find and update it.

**Review by mkflow27 — APPROVED** [review]

**Pull request merged**